containers/podman-desktop
# feat: nav sections

### What does this PR do?

Nav sections component. Every section starts collapsed and can be expanded to show a set of child nav items. We didn't have a complete design for this to I've tried a few things and ran them past Mairin briefly. State changes are animated, and selection styling updated to use purple outlines instead of left border.

Switched Docker Desktop extensions to use a section (and so that there is something to easily test here), using a new puzzle icon from Mairin (FA only has a solid puzzle). Kubernetes will be done in another PR.

This doesn't directly solve the problem that you could have so many icons that it doesn't fit onscreen, but as we have more sections they start closed and you can at least close most of them to help make sure the thing you need is visible.

### Screenshot / video of UI

https://github.com/containers/podman-desktop/assets/19958075/df73ef69-5340-41fb-92bb-f7e14455242c

### What issues does this PR fix or reference?

N/A

### How to test this PR?

Install at least one docker desktop extension. Expand/collapse to see how it looks.
